src/Symfony/Component/AssetMapper/Compressor/GzipCompressor.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\Compressor;
+
+use Symfony\Component\Process\ExecutableFinder;
+use Symfony\Component\Process\Process;
+
+/**
+ * Compresses a file using gzip.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class GzipCompressor implements CompressorInterface
+{
+    public function __construct(
+        private ?string $executable = null,
+    ) {
+    }
+
+    public function compress(string $path): void
+    {
+        if (file_exists("$path.gz")) {
+            return;
+        }
+
+        if (\in_array('compress.zlib', stream_get_wrappers(), true)) {
+            $this->compressWithExtension($path);
+
+            return;
+        }
+
+        $this->compressWithBinary($path);
+    }
+
+    private function compressWithExtension(string $path): void
+    {
+        if (!copy($path, "compress.zlib://$path.gz", stream_context_create(['zlib' => ['level' => 9]]))) {
+            throw new \RuntimeException(\sprintf('The compressed file "%s.gz" could not be written.', $path));
+        }
+    }
+
+    private function compressWithBinary(string $path): void
+    {
+        if (null === $this->executable) {
```

Comment:
This method needs to check whether the process component is available before using it, as it is not a requirement of the packages (same in other compressors).

Reply:
done